### DISCUSSION TITLE
How to add `"types": ["vitest/globals"]` only for test `**/*.spec.ts` files

### DISCUSSION THREAD

**Discussion by SeyyedKhandon** [vitest-dev/vitest#5086] [Q&A]
Currently we have enabled the usage of `describe, test, it, expect` etc. in our project via:

```
// vitest.config.js
/// <reference types="vitest" />

import { defineConfig, mergeConfig } from 'vitest/config'
import viteConfig from './vite.config.js'
import vue from '@vitejs/plugin-vue'

export default mergeConfig(
  viteConfig,
  defineConfig({
    plugins: [vue()],
    test: {
      setupFiles: ['./test/setup.ts'],
      globals: true,
      environment: 'jsdom',
    },
  })
)

```

```
//tsconfig.json
{
  "extends": "./.nuxt/tsconfig.json",
  "ts-node": {
    "compilerOptions": {
      "module": "commonjs",
      "strictNullChecks": true,
      "exactOptionalPropertyTypes": true
    }
  },
  "exclude": ["node_modules/**/*", "coverage/**/*"],
  "compilerOptions": {
    "strict": false,
    "esModuleInterop": true,
    "noUnusedLocals": true,
    "strictNullChecks": true,
    "exactOptionalPropertyTypes": false,
    "noErrorTruncation": true, // expand all component props on hover in vscode
    "types": ["vitest/globals"]
  }
}

```

The problem is we can use `describe, expect` etc. inside non test files, and typescript wont compain about it. How can we restrict this vitest globals only to test files?

ps. our test files are living next to our development files, therefore we don't have a separate `/test` folder.

**Comment by AriPerkkio** [comment]
This is more of a Typescript question rather than Vitest specific one. Here's discussion from Typescript: https://github.com/microsoft/TypeScript/issues/37053.

In summary, you can't. One of my favorite part of Vitest is that it does not pollute global typings by default like some test runners do. Importing `test` and `describe` in test files and disabling `globals` works always.

**Reply by devinrhode2** [reply]
For the tragically test-resistant devs out there, what does disabling globals look like?

**Reply by sheremet-va** [reply]
> For the tragically test-resistant devs out there, what does disabling globals look like? 

https://vitest.dev/config/#globals

**Comment by Habibulloh08** [comment]
```  /// <reference types="vitest" />
/// <reference types="vite/client" />
import { defineConfig } from "vite";
import react from "@vitejs/plugin-react";

// https://vitejs.dev/config/
export default defineConfig({
  plugins: [react()],
  test: {
    globals: true,
    environment: "jsdom",
    setupFiles: "./src/tests/setup.ts",
  },
});
```
